### DISCUSSION TITLE
copyq copy command not showing in window

### DISCUSSION THREAD

**Discussion by gbcox** [hluk/CopyQ#2208] [Q&A] [answered]
I'm trying to use copyq copy in a script to put variables in the clipboard.  However, when displaying the main window in KDE, the copied item does not show up.  If I enter "CTRL+SHIFT+V" in a konsole window it appears to be there, but again doesn't show up in the window.  When I try this in GNOME I don't know where it goes.  CTRL+SHIFT+V doesn't paste anything and it again doesn't show up in the main copyq window.  

I've tried just entering the commands from a terminal window instead of in a script and same issue.  I'm not at all familiar with GNOME, so am I missing something there to get copyq to work properly ?  Does the copyq copy command not put things in the copyq main window?  

Thanks!
CopyQ Clipboard Manager 6.3.2
Qt: 5.15.6
KNotifications: 5.98.0
Compiler: GCC
Arch: x86_64-little_endian-lp64
OS: Fedora Linux 37 (Workstation Edition)

OS: Fedora Linux 37 (Workstation Edition) x86_64 
Host: 80K4 Lenovo ideapad 500-15ACZ 
Kernel: 6.0.12-300.fc37.x86_64 
Uptime: 35 mins 
Packages: 3386 (rpm) 
Shell: bash 5.2.9 
Resolution: 1920x1080 
DE: Plasma 5.26.4 
WM: kwin 
Theme: Breeze [Plasma], Adwaita [GTK3] 
Icons: [Plasma], breeze [GTK2/3] 
Terminal: konsole 
CPU: AMD A10-8700P Radeon R6 4C+6G (4) @ 1.800GHz 
GPU: AMD ATI Radeon R5/R6/R7 Graphics 
Memory: 3849MiB / 7403MiB

**Comment by hluk** [comment]
> I'm trying to use copyq copy in a script to put variables in the clipboard. However, when displaying the main window in KDE, the copied item does not show up.

`copyq copy ...` would only put stuff into clipboard, but not in the clipboard history (use `copyq add ...` for that).

> If I enter "CTRL+SHIFT+V" in a konsole window it appears to be there, but again doesn't show up in the window. When I try this in GNOME I don't know where it goes. CTRL+SHIFT+V doesn't paste anything and it again doesn't show up in the main copyq window.

I did not use KDE Plasma in Fedora 37 much (something started eating CPU cycles so I switched to sway again) but I believe CopyQ worked well with the clipboard (and even with mouse selections) since Plasma provides the clipboard protocol that apps can use. Unfortunately, I think this is still not implemented in any way in GNOME.

Does CTRL+SHIFT+V paste clipboard or mouse selection? In addition to `copyq copy ...`, you may also need to use `copyq copySelection ...` since some terminals use the selection by default.

**Comment by gbcox** [comment]
>copyq copy ... would only put stuff into clipboard, but not in the clipboard history (use copyq add ... for that).

Thanks, that makes sense, and for the purposes of the script is fine.  I obviously keep missing that when reviewing the docs.  Sorry.

>I did not use KDE Plasma in Fedora 37 much (something started eating CPU cycles so I switched to sway again) but I believe CopyQ worked well with the clipboard (and even with mouse selections) since Plasma provides the clipboard protocol that apps can use. Unfortunately, I think this is still not implemented in any way in GNOME.

Yes, I haven't had any issues at all with KDE AFAIK.  The only thing I noticed was that entries were not showing up in the clipboard history, and as you pointed out that is because I'm using `copyq copy` instead of `copyq add` - and as I mentioned for the purposes of this script that is fine.

>Does CTRL+SHIFT+V paste clipboard or mouse selection? In addition to copyq copy ..., you may also need to use copyq copySelection ... since some terminals use the selection by default.

No problems with KDE, but GNOME gives me issues, and it doesn't help that I rarely use GNOME :-(
I've tried with Terminal and Konsole under GNOME with both `copyq copy` and `copyq copySelection`.  copyq responds with `true`, but it doesn't get placed in the clipboard, not sure where it goes.  `copyq add` puts it in clipboard history, and if I then go into the copyq gui and activate it, it places it in the clipboard.  Is there a way to activate from the command line?  Any clues to what might be happening? 

I've noticed if I `CTRL+SHIFT+C` that appears to be places in some other GNOME clipboard, just not copyq.  Am I missing something to tell GNOME to use copyq instead of what appears to be the default GNOME clipboard?

**Reply by hluk** [reply]
Not sure what's happening. I would expect at least copying to work in GNOME/Wayland if not clipboard monitoring.

Is there any error/warning in logs (F12 shortcut from the main window)?

**Reply by gbcox** [reply]
> Not sure what's happening. I would expect at least copying to work in GNOME/Wayland if not clipboard monitoring.
> 
> Is there any error/warning in logs (F12 shortcut from the main window)?

Yeah, it's weird.  That is why I was thinking I might have to do something else to get it working under GNOME.  Here is what the log says:
Warning [2023-01-04 07:17:49.231] <Server-7839>: [default] QtWarning: Activating Wayland clipboard took 5000 ms
ERROR [2023-01-04 07:17:49.237] <Server-7839>: [default] QtCritical: Failed to activate Wayland clipboard
Warning [2023-01-04 07:17:54.580] <monitorClipboard-8900>: [default] QtWarning: Activating Wayland clipboard took 5000 ms
ERROR [2023-01-04 07:17:54.589] <monitorClipboard-8900>: [default] QtCritical: Failed to activate Wayland clipboard
Warning [2023-01-04 07:22:26.892] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:10.687] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:10.695] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:10.742] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:10.750] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:11.483] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:11.490] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:15.215] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:15.222] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:16.155] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:16.162] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:16.890] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:16.897] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:17.058] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:17.065] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:17.373] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:17.381] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:17.497] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:17.506] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:17.535] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:17.544] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:17.558] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()
Warning [2023-01-04 07:23:17.566] <Server-7839>: [qt.qpa.wayland] QtWarning: Wayland does not support QWindow::requestActivate()

**Reply by hluk** [reply]
I guess the only workaround for **CopyQ to work in GNOME** is to start the app in X11 compatibility mode (XWayland):

```bash
env QT_QPA_PLATFORM=xcb copyq
```

**Reply by gbcox** [reply]
> I guess the only workaround for **CopyQ to work in GNOME** is to start the app in X11 compatibility mode (XWayland):
> 
> ```shell
> env QT_QPA_PLATFORM=xcb copyq
> ```

ok, I'll make that change in the script for the gnome wayland environment.  If something changes LMK.  In the meantime, I'll mark this as answered.  Thanks again!

**Comment by gbcox** [comment, marked as answer]
https://github.com/hluk/CopyQ/discussions/2208#discussioncomment-4599433
